From 48e31f9fc3cf3c486c3d27a67b2687f6df0c5c71 Mon Sep 17 00:00:00 2001
From: Denis Kenzior <denkenz@gmail.com>
Date: Thu, 16 May 2019 15:10:53 -0500
Subject: main: Quiet ld errors with external ell

When oFono is built with --enable-external-ell, the compiler for some
reason does not generate a debug section on some systems.  This is due
to the fact that l_debug is never called.  However, ell also does not
call l_debug, yet when built-in ell is used, the section is created by
the compiler.

For now work around this by adding a no-op l_debug() call in main.c.
The real fix is to migrate all of the oFono logging functionality to use
ell instead.

Upstream-Status: Backport
Signed-off-by: Adrian Bunk <bunk@stusta.de>
---
 src/main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/main.c b/src/main.c
index 8623a060..4529cde1 100644
--- a/src/main.c
+++ b/src/main.c
@@ -236,6 +236,7 @@ int main(int argc, char **argv)
 	event_loop = g_main_loop_new(NULL, FALSE);
 
 	l_log_set_stderr();
+	l_debug("");
 	l_debug_enable("*");
 	l_main_init();
 
-- 
2.20.1

